lib/configSchema.js

```diff
@@ -125,6 +125,15 @@ const schema = {
   additionalProperties: false,
   required: ['provider', 'service'],
   definitions: {
+    // TODO: awsKmsArn & awsKmsArnString definitions to be moved to lib/plugins/aws/provider/awsProvider.js once service.awsKmsKeyArn moved to provider.awsKmsKeyArn, see https://github.com/serverless/serverless/issues/8261
+    // TODO: awsKmsArn to include #/definitions/awsCfFunction instead of type: object as one of the possible definition, see https://github.com/serverless/serverless/issues/8261
+    awsKmsArn: {
+      anyOf: [{ type: 'object' }, { $ref: '#/definitions/awsKmsArnString' }],
+    },
+    awsKmsArnString: {
+      type: 'string',
+      pattern: '^arn:aws[a-z-]*:kms',
+    },
```

Comment:
I think we don't need this definition on its own? I would just put this pattern into `awsKmsArn`